airflow/operators/python.py

```diff
@@ -450,6 +451,12 @@ def __init__(
             raise ValueError(f"{type(self).__name__} only supports functions for python_callable arg")
         if inspect.isgeneratorfunction(python_callable):
             raise ValueError(f"{type(self).__name__} does not support using 'yield' in python_callable")
+        if env_vars:
+            for key, value in env_vars.items():
+                if not isinstance(key, str):
+                    raise ValueError(f"env_vars key ${key} must be a string, but got {type(key)}")
+                if not isinstance(value, str):
+                    raise ValueError(f"env_vars value must be a string, but got {type(value)}")
```

Comment:
As a general rule, argument validation should be done in `execute` instead, to accomodate possible user customisation such as policy rules.

I kind of feel we don’t actually need to do these checks at all since non-string values will cause execution to eventually fail anyway.

Reply:
@uranusjr 
Oh, I forgot the policy!
As you mentioned, errors will be automatically raised when passed to Popen.
```python
execute_in_subprocess_with_kwargs(["bash", "-c", "export"], env={"MY_VARS": 1})
# output:
# File "/usr/lib/python3.10/subprocess.py", line 971, in __init__
#     self._execute_child(args, executable, preexec_fn, close_fds,
#   File "/usr/lib/python3.10/subprocess.py", line 1783, in _execute_child
#     env_list.append(k + b'=' + os.fsencode(v))
#   File "/usr/lib/python3.10/os.py", line 811, in fsencode
#     filename = fspath(filename)  # Does type-checking of `filename`.
# TypeError: expected str, bytes or os.PathLike object, not int
```
I will make the changes